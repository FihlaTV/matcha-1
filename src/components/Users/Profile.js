import React from 'react';
import { ScrollView, StyleSheet, View, Image } from 'react-native';
import type { UserType } from 'src/types/User';
import { COLORS, MARGINS } from 'src/constants/design';
import { User } from 'src/components/Users';
import type { PhotoType } from 'src/types/Photo';
import LoadingHOC from 'src/components/HOC/LoaderHOC';

const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.BACKGROUND,
    flex: 0,
    margin: MARGINS.TINY,
  },
  photoImageThumbnail: {
    backgroundColor: COLORS.GREY_DARK,
    height: 100,
    margin: MARGINS.TINY,
    width: 100,
  },
  photosContainer: {
    borderColor: COLORS.GREY_LIGHTEST,
    borderWidth: 1,
    flex: 1,
    flexDirection: 'row',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    width: 347,
  },
});

type PhotoThumbnailProps = {
  photo: PhotoType,
};

const PhotoThumbnail = (props: PhotoThumbnailProps) => (
  <Image style={styles.photoImageThumbnail} source={{ uri: props.photo.url }} />
);

type PhotoThumbnailsProps = {
  photo: PhotoType,
};

const PhotoThumbnails = (props: PhotoThumbnailsProps) => (
  props.photos.length !== 0 &&
  <View style={styles.photosContainer}>
    {
      props.photos.map((photo) => (<PhotoThumbnail key={photo.id} photo={photo} />))
    }
  </View>
);

const ExtendedPhotoThumbnails = LoadingHOC('photos')(PhotoThumbnails);

type Props = {
  user: UserType,
  photos: Array<PhotoType>,
};

const Profile = (props: Props) => (
  <ScrollView style={styles.container} contentContainerStyle={{ alignItems: 'center' }}>
    <User user={props.user} />
    <ExtendedPhotoThumbnails photos={props.photos} />
  </ScrollView>
);

export default Profile;
