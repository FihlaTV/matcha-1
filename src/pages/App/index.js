import React from 'react';
import { View } from 'react-native';
import Footer from 'app/components/Footer';
import Layout from 'app/components/Layout/Basic';

const App = () => (
  <Layout>
    <View style={{ flex: 1 }} />
    <Footer />
  </Layout>
);

export default App;
export { App };
