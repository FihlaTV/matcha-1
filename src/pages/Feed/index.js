import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ScrollView } from 'react-native';
import Layout from 'src/components/Layout/Basic';
import Photos from 'src/components/Photos';
import { onLoadPhotos } from 'src/pages/Feed/actions';
import { selectPhotos } from 'src/pages/Feed/selector';
import type { Photo } from 'src/types/Photo';

type Props = {
  photos: Array<Photo>,
  onLoadPhotos: Function,
};

class Feed extends Component<Props> {
  componentWillMount() {
    this.props.onLoadPhotos();
  }

  render() {
    return (
      <Layout>
        <ScrollView style={{ flex: 1 }}>
          <Photos photos={this.props.photos} />
        </ScrollView>
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  photos: selectPhotos(state),
});

const mapDispatchToProps = {
  onLoadPhotos,
};

export default connect(mapStateToProps, mapDispatchToProps)(Feed);
