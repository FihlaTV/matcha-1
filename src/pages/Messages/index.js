import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ScrollView } from 'react-native';
import Layout from 'app/components/Layout/Basic';
import Threads from 'app/components/Threads';
import { onLoadThreads } from 'app/pages/Messages/actions';
import { selectThreads } from 'app/pages/Messages/selector';
import type { Thread } from 'app/types/Thread';

type Props = {
  threads: Array<Thread>,
};

class Messages extends Component<Props> {
  componentWillMount() {
    this.props.onLoadThreads();
  }

  render() {
    return (
      <Layout>
        <ScrollView style={{ width: 300 }}>
          <Threads threads={this.props.threads} />
        </ScrollView>
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  threads: selectThreads(state),
});

const mapDispatchToProps = {
  onLoadThreads,
};

export default connect(mapStateToProps, mapDispatchToProps)(Messages);
export { Messages };
