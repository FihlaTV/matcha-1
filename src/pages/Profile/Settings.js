import React, { Component } from 'react';
import { connect } from 'react-redux';
import { StyleSheet, Text, View, Button } from 'react-native';
import { injectIntl } from 'react-intl';
import { MARGINS, FONT_SIZES, FIELD_HEIGHT } from 'src/constants/design';
import { signUp } from 'src/pages/Auth/actions';
import Layout from 'src/components/Layout/Basic';
import TextField from 'src/primitives/TextField';

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    height: 7 * FIELD_HEIGHT, // eslint-disable-line
    justifyContent: 'space-around',
    width: '80%',
  },
  title: {
    fontSize: FONT_SIZES.BIG,
    marginBottom: MARGINS.BIG,
  },
});

type Props = {
  intl: any,
  signUp: Function,
};

type State = {
  email: string,
  name: string,
  password: string,
  username: string,
}

class Settings extends Component<Props, State> {
  state = {
    email: '',
    name: '',
    password: '',
    username: '',
  }

  render() {
    return (
      <Layout>
        <View style={styles.container}>
          <Text style={styles.title}>{this.props.intl.formatMessage({ id: 'SettingsPage.title' })}</Text>
          <TextField
            placeholder={this.props.intl.formatMessage({ id: 'SettingsPage.username' })}
            title={this.props.intl.formatMessage({ id: 'SettingsPage.username' })}
            value={this.state.username}
            autoCapitalize="none"
            onChangeText={(text) => this.setState({ username: text })}
          />
          <TextField
            placeholder={this.props.intl.formatMessage({ id: 'SettingsPage.name' })}
            value={this.state.name}
            onChangeText={(text) => this.setState({ name: text })}
          />
          <TextField
            placeholder={this.props.intl.formatMessage({ id: 'SettingsPage.email' })}
            value={this.state.email}
            autoCapitalize="none"
            onChangeText={(text) => this.setState({ email: text })}
            keyboardType="email-address"
          />
          <TextField
            placeholder={this.props.intl.formatMessage({ id: 'SettingsPage.password' })}
            value={this.state.password}
            autoCapitalize="none"
            secureTextEntry
            onChangeText={(text) => this.setState({ password: text })}
          />
          <Button
            title={this.props.intl.formatMessage({ id: 'SettingsPage.signUp' })}
            onPress={() => this.props.signUp({
              activeAt: null,
              createdAt: null,
              dob: null,
              email: this.state.email,
              lat: 0,
              long: 0,
              name: this.state.name,
              password: this.state.password,
              picture: '',
              subscribed: false,
              username: this.state.username,
            })}
          />
        </View>
      </Layout>
    );
  }
}

const mapDispatchToProps = {
  signUp,
};

export default connect(null, mapDispatchToProps)(injectIntl(Settings));
