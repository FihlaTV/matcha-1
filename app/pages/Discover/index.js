import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ScrollView } from 'react-native';
import Layout from 'app/components/Layout/Basic';
import Users from 'app/components/Users';
import { onLoadUsers } from 'app/pages/Discover/actions';
import { selectUsers } from 'app/pages/Discover/selector';
import type { UserType } from 'app/types/User';
import { selectCurrentUser } from 'app/pages/Auth/selector';

type Props = {
  users: Array<UserType>,
  currentUser: UserType,
};

class Discover extends Component<Props> {
  componentWillMount() {
    this.props.onLoadUsers();
  }

  render() {
    return (
      <Layout>
        <ScrollView style={{ flex: 1 }}>
          <Users users={this.props.users} currentUser={this.props.currentUser} />
        </ScrollView>
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  currentUser: selectCurrentUser(state),
  users: selectUsers(state),
});

const mapDispatchToProps = {
  onLoadUsers,
};

export default connect(mapStateToProps, mapDispatchToProps)(Discover);
export { Discover };
