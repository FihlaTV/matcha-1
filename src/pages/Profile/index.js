import React, { Component } from 'react';
import { connect } from 'react-redux';
import Layout from 'src/components/Layout/Basic';
import { selectCurrentUser } from 'src/pages/Auth/selector';
import type { UserType } from 'src/types/User';
import { onLoadPhotosUser } from 'src/pages/Feed/actions';
import ProfileComponent from 'src/components/Users/Profile';
import { selectPhotosUser } from 'src/pages/Feed/selector';
import type { PhotoType } from 'src/types/Photo';

type Props = {
  currentUser: UserType,
  onLoadPhotosUser: Function,
  photos: Array<PhotoType>,
};

class Profile extends Component<Props> {
  componentWillMount() {
    this.props.onLoadPhotosUser(this.props.currentUser.id);
  }

  render() {
    return (
      <Layout style={{ flexDirection: 'row', justifyContent: 'center' }}>
        <ProfileComponent user={this.props.currentUser} photos={this.props.photos} />
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  currentUser: selectCurrentUser(state),
  photos: selectPhotosUser(state),
});

const mapDispatchToProps = {
  onLoadPhotosUser,
};

export default connect(mapStateToProps, mapDispatchToProps)(Profile);
