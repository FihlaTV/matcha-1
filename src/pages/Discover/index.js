// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { ScrollView } from 'react-native';
import Layout from 'src/components/Layout/Basic';
import Users from 'src/components/Users';
import { onLoadUsers, type Action } from 'src/pages/Discover/actions';
import { selectUsers } from 'src/pages/Discover/selector';
import { selectCurrentUser } from 'src/pages/Auth/selector';

type Props = {
  users: Array<UserType>,
  currentUser: UserType,
  onLoadUsers: () => Action,
};

class Discover extends Component<Props> {
  componentWillMount() {
    this.props.onLoadUsers();
  }

  render() {
    return (
      <Layout>
        <ScrollView style={{ flex: 1 }}>
          <Users users={this.props.users} currentUser={this.props.currentUser} />
        </ScrollView>
      </Layout>
    );
  }
}

const mapStateToProps = (state) => ({
  currentUser: selectCurrentUser(state),
  users: selectUsers(state),
});

const mapDispatchToProps = {
  onLoadUsers,
};

export default connect(mapStateToProps, mapDispatchToProps)(Discover);
