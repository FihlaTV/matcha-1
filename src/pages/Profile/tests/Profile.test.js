import React from 'react';
import { Profile } from 'app/pages/Profile';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Profile', () => {
  describe('render test', () => {
    it('should render', () => {
      const wrapper = shallow(<Profile />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});
