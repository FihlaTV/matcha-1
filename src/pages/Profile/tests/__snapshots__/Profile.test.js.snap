// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile render test should render 1`] = `
<BasicLayout
  hasBack={false}
  headerActions={Array []}
  noTabs={false}
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
    }
  }
>
  <ScrollView
    style={
      Object {
        "backgroundColor": "white",
        "flex": 0,
        "margin": 5,
      }
    }
  >
    <User />
  </ScrollView>
</BasicLayout>
`;
